lib/arborist/reify.js

```diff
@@ -905,7 +905,7 @@ module.exports = cls => class Reifier extends cls {
           // would allow versions outside the requested range.  Tags and
           // specific versions save with the save-prefix.
           const isRange = (subSpec || req).type === 'range'
-          const range = !isRange || subset(prefixRange, spec, { loose: true })
+          const range = !isRange || subset(prefixRange, spec, { loose: true }) || spec === '*'
```

Comment:
If this is the route to go, thoughts on switching these checks for speed?

```suggestion
          const range = !isRange || spec === '*' || subset(prefixRange, spec, { loose: true })
```